python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -119,6 +121,77 @@ def categorical_embedding_with_indices(feature_tensor, feature_info, file_io: Fi
     return embedding
 
 
+def categorical_embedding_to_encoding_bilstm(feature_tensor, feature_info, file_io: FileIO):
```

Comment:
@Ullimague Can we refactor this to do only categorical embedding on indices and then have another function to convert categories to indices from vocabulary file? This would allow us to reuse the same function for the char embedding in `sequence.py` as well.

Reply:
Refactored: extracted vocabulary read and bilstm part @lastmansleeping